### PR TITLE
Add argo workflow

### PR DESCRIPTION
**Pull request by mar4uk**
Fixes https://github.com/grafana/otel-squad/issues/24
Added drone to enable ARGO workflow.
Because this repo is public we can’t trigger ARGO workflow from github actions (grafana org secrets for Argo are not available for public repos). That’s why drone was added

I kept github actions because it was not easy to rewrite the logic goreleaser does to drone. Probably we can rewrite it later (or keep goreleaser to build packages, archives, and publish docker images for multi architectures as it is)

In the current implementation drone is used for push event to main branch to:
- run tests
- build and push `otel-grafana` docker image for linux/amd64 to ghcr.io registry
- trigger argo workflow

### CODE REVIEW

File: .drone/drone.jsonnet [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  _repo(app):: $._registry + '/' +'grafana/opentelemetry-collector-components/' + app,
+  _registry:: 'ghcr.io',
+};
+
+[
+  pipeline('check')
+  + withStep(check.step)
+  + triggers.main,
+
+  pipeline('check gcom api processor')
```
Comment (kovrus) [inline .drone/drone.jsonnet:115]:
```suggestion
  pipeline('test gcom api processor')
```

File: .drone/drone.jsonnet [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      image='us.gcr.io/kubernetes-dev/drone/plugins/argo-cli'
+    )
+  )
+  + withImagePullSecrets
+  + triggers.main,
+]
++ [
+  vault.secret('dockerconfigjson', 'secret/data/common/gcr', '.dockerconfigjson'),
+  vault.secret('gh_token', 'infra/data/ci/github/grafanabot', 'pat'),
+  vault.secret('gcr_admin', 'infra/data/ci/gcr-admin', 'service-account'),
```
Comment (kovrus) [inline .drone/drone.jsonnet:157]: is it used anywhere?

Comment (mar4uk) [reply .drone/drone.jsonnet]: it is not, thank you! removed

**Review by kovrus — APPROVED** [review]
lgtm

Cannot due a very thoughtful review due lack of drone knowledge

**Pull request merged**